### PR TITLE
Update knobs to controls in storybook: SelectQuotePopover

### LINKED ISSUE #18500: Update knobs to controls in storybook: SelectQuotePopover
### Description
Currently there are a few storybook stories still using knobs in our storybook files `*.stories.js`. knobs have been deprecated in favour of controls. It would be great to migrate these last stories from knobs to controls.

This issue is to migrate the `ui/pages/swaps/select-quote-popover/select-quote-popover.stories.js` from knobs to controls

Screenshot 2023-04-06 at 4 55 49 PM

### Requirements
If you would like to take on this issue please comment but before asking to take on this issue there are a few requirements that will help complete this task:
- Get storybook up and running `yarn storybook`
- Some knowledge of storybook is not required but would be beneficial for this task. [storybook docs here](https://storybook.js.org/)

### Technical Details
- Migrate knobs to controls

### Acceptance Criteria
- knobs have been migrated to controls for listed story
- Take Before / After screenshots of the knobs and new controls and attach to the PR
- Tag @georgewrmarshall, @garrettbear, or @NidhiKJha in the PR to review

### PR DESCRIPTION
**Pull request by sumitshinde-84**
## Explanation
We have updated the argTypes of the story to use the control function for the quoteDataRows property instead of the object function. This is because the object function has been deprecated and replaced with the control function for object-type properties in Storybook 6.3 and above. The control function with type: 'object' will render an object control that allows the user to edit the value of the object property. We have also updated the usage of quoteDataRows in the component to use args.quoteDataRows instead of the object knob. This ensures that the value passed to the component is the value of the quoteDataRows prop from the args object, instead of the value returned by the object knob

### Before
beforePopover

### After
WhatsApp Image 2023-04-20 at 1 06 44 AM (1)
WhatsApp Image 2023-04-20 at 1 06 44 AM (2)

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [x] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [x] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
closes #18500

### CODE REVIEW

File: ui/pages/swaps/select-quote-popover/select-quote-popover.stories.js [hunk 1]
```diff
@@ -1,6 +1,5 @@
 import React, { useState } from 'react';
 import { action } from '@storybook/addon-actions';
```
Comment (georgewrmarshall) [inline ui/pages/swaps/select-quote-popover/select-quote-popover.stories.js:2]: Would it be possible to use the actions that come from controls here? In argTypes you could do something like 
```
 onSubmit: { action: 'onSubmit' },
```

**Review by georgewrmarshall — COMMENTED** [review]
Hey @sumitshinde-84, this looks great and fantastic PR description! I left one suggestion regarding action controls

**Comment by sumitshinde-84** [comment]
I have made the changes as you suggested. Please review them and let me know if there are any mistakes or improvements that can be made.

**Comment by sumitshinde-84** [comment]
ci/circleci: test-deps-audit  is consistently failing , To resolve the issue, you should investigate the details of the vulnerability, including the severity level and the module(s) affected. Then, you should take the necessary steps to fix the vulnerability, such as updating the affected module(s) to a non-vulnerable version, applying a security patch, or removing the vulnerable code altogether.

File: ui/pages/swaps/select-quote-popover/select-quote-popover.stories.js [hunk 2]
```diff
@@ -29,11 +27,16 @@ export default {
...
-      control: 'object',
+      control: {
+        type: 'object',
+      },
     },
     hideEstimatedGasFee: {
       control: 'boolean',
     },
+    onSubmit: {
+      action: 'submit SelectQuotePopover',
```
Comment (georgewrmarshall) [inline ui/pages/swaps/select-quote-popover/select-quote-popover.stories.js:38]: Our convention for actions is just to comment the prop
```suggestion
      action: 'onSubmit',
```

**Review by georgewrmarshall — COMMENTED** [review]
Looking good! Left one suggestion

**Review by georgewrmarshall — APPROVED** [review]
LGTM!

**Review by NidhiKJha — APPROVED** [review]

**Pull request merged**